src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFChart.java

```diff
@@ -631,7 +695,19 @@ public XSSFWorkbook getWorkbook() throws IOException, InvalidFormatException {
      * @since POI 4.0.0
      */
     public void setWorkbook(XSSFWorkbook workbook) {
-        this.workbook = workbook;
+        File file;
+        FileOutputStream fos;
+        try {
+            file = TempFile.createTempFile("TempEmbedded",".xlsx");
+            fos = new FileOutputStream(file);
+            workbook.write(fos);
+            fos.close();
+            this.workbook = new XSSFWorkbook(file);
```

Comment:
Why writing the given workbook to a file?
Why using a file which is outside of the OPC package?
Why creating a new workbook from the same content?

Reply:
i tried to use existing workbook and we are able to successfully update workbook sheet data but at the time of writing embedded part it is writing old input-stream data and due to that when we are modifying any property from word/ppt it showing old data which we read at the time of reading  word/ppt file. so what i found the input-stream writing old data and due to that i first write old data into temp-file and then read that file.

Reply:
This code was trying to work around some bug in XSSFWorkbook. Now the workaround has been officially committed to the trunk. According to recent conversation on the POI Developers List, the origin of the issue with not correctly writing only the new values in the workbook has been fixed, so that this work around would not be required anymore.

Reply:
I undid this on trunk just now (tests still pass)